File: perl/Grace.pl
#!/usr/bin/env perl

use strict;
use warnings;

#
#	perl!
#

my $PROGRAM = 'open my $f, ">", "Grace_kid.pl" or die; printf $f "$FORMAT", 10, 9, 39, "$FORMAT";';
my $FORMAT = '#!/usr/bin/env perl%1$c%1$cuse strict;%1$cuse warnings;%1$c%1$c#%1$c#%2$cperl!%1$c#%1$c%1$cmy $PROGRAM = %3$copen my $f, ">", "Grace_kid.pl" or die; printf $f "$FORMAT", 10, 9, 39, "$FORMAT";%3$c;%1$cmy $FORMAT = %3$c%4$s%3$c;%1$c%1$ceval $PROGRAM;%1$c';

eval $PROGRAM;

File: python/Grace.py
#!/usr/bin/env python3

FORMAT = "#!/usr/bin/env python3{1}{1}FORMAT = {0}{2}{0}{1}PROGRAM = {0}with open('Grace_kid.py', 'w') as f: print(FORMAT.format(chr(34), chr(10), FORMAT), file=f){0}{1}MAIN = exec{1}{1}{0}{0}{0}{1}    no macros? :({1}{0}{0}{0}{1}{1}MAIN(PROGRAM)"
PROGRAM = "with open('Grace_kid.py', 'w') as f: print(FORMAT.format(chr(34), chr(10), FORMAT), file=f)"
MAIN = exec

"""
    no macros? :(
"""

MAIN(PROGRAM)
